metrics-core/src/main/java/com/codahale/metrics/SlidingTimeWindowMovingAverages.java

```diff
@@ -39,7 +38,7 @@ public class SlidingTimeWindowMovingAverages implements MovingAverages {
      * One counter per time bucket/slot (i.e. per second, see TICK_INTERVAL) for the entire
      * time window (i.e. 15 minutes, see TIME_WINDOW_DURATION_MINUTES)
      */
-    private ArrayList<LongAdder> buckets;
+    private final LongAdder[] buckets;
```

Comment:
a separate optimization would be to avoid the `LongAdder`s and use `AtomicLongArray`

```suggestion
    private final AtomicLongArray buckets;
```

Reply:
using `AtomicLongArray` cuts optimized `getM1Rate` from ~60ns to ~45ns
```
Benchmark                          (recordings)                                    (type)  Mode  Cnt     Score    Error  Units
MovingAverageBenchmarks.getM1Rate            10           SlidingTimeWindowMovingAverages  avgt   20  1366.241 ± 24.341  ns/op
MovingAverageBenchmarks.getM1Rate            10  OptimizedSlidingTimeWindowMovingAverages  avgt   20    43.180 ±  0.153  ns/op
MovingAverageBenchmarks.getM1Rate          1000           SlidingTimeWindowMovingAverages  avgt   20  1377.545 ± 65.159  ns/op
MovingAverageBenchmarks.getM1Rate          1000  OptimizedSlidingTimeWindowMovingAverages  avgt   20    43.967 ±  2.058  ns/op
```